website/docs/reference/snapshot-configs.md

```diff
@@ -150,11 +145,25 @@ Configurations can be applied to snapshots using the [YAML syntax](/docs/build/s
 
 </Tabs>
 
+### Snapshot configuration migration
+
+The latest snapshot configurations introduced in dbt v1.9 (such as [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names), [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current), and `hard_deletes`) are best suited for new snapshots. For existing snapshots, we recommend the following to avoid any inconsistencies in your snapshots:
+
+#### For existing snapshots
+- Migrate tables &mdash; Migrate the previous snapshot to the new table schema and values:
+  - Create a backup copy of your snapshots.
+  - Use `alter` statements as needed (or a script to apply `alter` statements) to ensure table consistency.
+- New configurations &mdash; Convert the configs one at a time, testing as you go. 
+
+:::warning
+If you use one of the latest configs, such as `dbt_valid_to_current`, without migrating your data, you may have mixed old and new data, leading to an incorrect downstream result.
+:::
```

Comment:
I think we should link to this section within the v1.9 upgrade guide:

website/docs/docs/dbt-versions/core-upgrade/06-upgrading-to-v1.9.md